base/applications/mspaint/atlimagedx.h

```diff
@@ -0,0 +1,137 @@
+#pragma once
+
+class CImageDx : public CImage
+{
+public:
+    CImageDx() : CImage()
+    {
+        GetImageHorizontalResolution = NULL;
+        GetImageVerticalResolution = NULL;
+        BitmapSetResolution = NULL;
+    }
+
+    BOOL GetResolution(Gdiplus::GpImage *pImage, float *pxDpi, float *pyDpi)
+    {
+        *pxDpi = 96;
+        *pyDpi = 96;
+
+        if (GetImageHorizontalResolution == NULL || GetImageVerticalResolution == NULL)
+        {
+            GetImageHorizontalResolution = AddrOf<GETIMAGEHORIZONTALRESOLUTION>("GdipGetImageHorizontalResolution");
+            GetImageVerticalResolution = AddrOf<GETIMAGEVERTICALRESOLUTION>("GdipGetImageVerticalResolution");
+        }
+
+        if (GetImageHorizontalResolution == NULL || GetImageVerticalResolution == NULL)
+            return FALSE;
+
+        GetImageHorizontalResolution(pImage, pxDpi);
+        GetImageVerticalResolution(pImage, pyDpi);
+        return TRUE;
+    }
+
+    BOOL SetResolution(Gdiplus::GpBitmap *pBitmap, float xDpi, float yDpi) const
+    {
+        if (xDpi <= 0)
+            xDpi = 96;
+        if (yDpi <= 0)
+            yDpi = 96;
```

Comment:
Why not just keeping whatever is given to the function? (Meaning also that your default values in SaveDx may need to be 96 instead.)

Reply:
Deleted.